enhancements/console/internationalization.md

```diff
@@ -0,0 +1,135 @@
+
+### Implementation Details/Notes/Constraints [optional]
+
+Localization will affect multiple teams within the console development space.
+This will affect Dev console, Admin console, and other packages like CNV.
+The solution must account for the plugin contribution model and enable dynamic extensions to provide localization as necessary.
+
+This is expected to include all information we can localize in the console, potentially broken over multiple phases.
+As mentioned earlier in this document, localization of all text within the system is not possible. In addition, some terms are Kubernetes terms and would not be translated.
+
+### Risks and Mitigations
+
+**Risk**: All text is not translated.
+
+**Mitigation**: Work with the UXD team on a design strategy.
+
+**Risk**: Translations are incorrect or don't work as expected.
+
+**Mitigation**: We will work with the Red Hat Globalization team to thoroughly test the application and obtain culturally sensitive and accurate translations.
+
+## Design Details
+
+### Open Questions [optional]
+
+* Will translations be done centrally for all plugins or will every team needs to have a separate team of translators?
+Setting up a team of translators specifically for kubevirt might be quite a challenge.
+* What is the desired fallback experience if not all pieces/parts of the UI are translated for a given language?
+* Can the user opt out of the language setting provided by the browser?
+* What will the size be for ongoing maintenance, support, and additional testing of localizations?
+* What capacity does the existing localization team have for additional product support?
+* Would every release of OpenShift be localized?
+
+### Test Plan
+
+We will have unit and integration tests. We will also work with the Globalization team to do localization testing.
+
+### Graduation Criteria
+
+Initially we will just mark strings internally. Translations and testing will be added once marking is complete.
+
+### Upgrade / Downgrade Strategy
+
+We will need to make sure that new components with hard-coded strings are marked for translation.
+
+## Implementation History
+
+* 2020-08-03 - [PR against console](https://github.com/openshift/console/pull/6194) opened
+* 2020-04-23 - [UXD exploration of i18next](https://github.com/jschuler/console/pull/1) opened
+* 2020-04-13 - [UXD analysis and comparison of libraries](https://github.com/patternfly/patternfly-react/issues/3952#issuecomment-613645855) posted
+
+## Drawbacks
+
+We will need to maintain accurate translations for every language we support and ensure that all new text in the front-end is considered for localization going forward. Testing and maintainence will also be a concern.
+
+## Alternatives
+
+Alternatives would be significantly worse - we could use different code bases for different languages or a different globalization framework (or skip localization entirely).
+
+We explored other globalization frameworks, and i18next was the best fit.
```

Comment:
I would change this section to mention react-intl with a link to https://github.com/patternfly/patternfly-react/issues/3952